File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/multiline/uncomment/direct/Java/Go/s573993837_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type InputReader struct {
	tokens []string
	idx    int
}

func NewInputReader(r *bufio.Reader) *InputReader {
	scanner := bufio.NewScanner(r)
	scanner.Split(bufio.ScanWords)
	var tokens []string
	for scanner.Scan() {
		tokens = append(tokens, scanner.Text())
	}
	return &InputReader{tokens, 0}
}

func (r *InputReader) Next() string {
	if r.idx < len(r.tokens) {
		r.idx++
		return r.tokens[r.idx-1]
	}
	return ""
}

func (r *InputReader) NextInt() int {
	num, _ := strconv.Atoi(r.Next())
	return num
}

func (r *InputReader) NextInt64() int64 {
	num, _ := strconv.ParseInt(r.Next(), 10, 64)
	return num
}

func min(a, b int64) int64 {
	if a < b {
		return a
	}
	return b
}

func solve(sc *InputReader, out *bufio.Writer) {
	t := sc.NextInt()
	for t > 0 {
		n := sc.NextInt64()
		m := sc.NextInt64()
		if n == 0 {
			fmt.Fprintln(out, 0)
			t--
			continue
		}
		if n == m {
			fmt.Fprintln(out, 0)
			t--
			continue
		}
		if n > m {
			rem := n % m
			res := min(rem, m-rem)
			fmt.Fprintln(out, res)
		} else {
			res := min(n, m-n)
			fmt.Fprintln(out, res)
		}
		t--
	}
	out.Flush()
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	sc := NewInputReader(reader)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()
	solve(sc, writer)
}

// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/mad/Java/Python/s573993837_commented.py
import sys
class InputReader:
    def __init__(self):
        self.tokens = []
        self.position = 0
        self.tokens = sys.stdin.read().split()
    def next(self):
        if self.position >= len(self.tokens):
            raise EOFError()
        self.position +=1
        return self.tokens[self.position-1]
    def nextInt(self):
        return int(self.next())
    def nextLong(self):
        return int(self.next())
    def nextDouble(self):
        return float(self.next())
def solve(sc, pw):
    t = 1
    for _ in range(t):
        n = sc.nextLong()
        m = sc.nextLong()
        if n == 0:
            print(0, file=pw)
            continue
        if n == m:
            print(0, file=pw)
            continue
        if n > m:
            n %= m
        res = min(m - n, n)
        print(res, file=pw)
def main():
    sc = InputReader()
    solve(sc, sys.stdout)
if __name__ == "__main__":
    main()
#End of Code
